docs/configuration.md

```diff
@@ -6,15 +6,49 @@ This document describes how to configure your epoch node installed using a relea
+**It is very important that a node not only can connect to other nodes, but that can accept incoming connections as well: the more peer connections (both inbound+outbound) the node has, the better its overall p2p network latency (i.e. block propagation time) will be.**
 
-(You do not need to specify the host at which your node can be contacted from the Internet, as each peer you ping will infer that from the address of the inbound TCP connection.)
+By default node listen on TCP port **3015**. It can be changed by `sync` > `port` parameter in the configuration file in case for some reason that port cannot be used (e.g. already used by other service).
 
-Please notice that, if your node is behind a firewall, you need to open a TCP port in your firewall (`sync` > `port` parameter) and map that port to the one the node actually listens on (`sync` > `port` parameter - the same). If the publicly available port needs to be different from the internal one, you need to set (`sync` > `external_port`) accordingly.
+#### Firewalls
 
-The following example configuration assumes that:
-* The listening TCP port on your public IP address is `3015`;
-* The listening TCP port on your node is `3115`.
+Example setup: `node (firewall) <-> firewall <-> Internet`
+
+If the node is behind a firewall that port  (default 3015 or the one set in the configuration) should be opened for inbound TCP connections.
+
+
+Note that the port may need to be opened both on the host machine running the node and any external device (firewall/router) that route the network traffic.
+Unfortunately all firewall configurations are different and common steps cannot be provided here, one should follow the documentation of their equipment/software "how to open firewall port".
+
+#### NAT
+
+Example setup: `node <-> NAT (router) <-> Internet`
+
+If the node is behind NAT (e.g. home router) the port should be forwarded on that device to accept incoming connections from Internet and route it to the node.
+
+Unfortunately all router configurations are different and common steps cannot be provided here, one should follow the documentation of their equipment/software "how to forward ports".
+
+##### Advanced NAT
+
+This is advanced configuration and should be used with caution because it can cause node misconfiguration and bad p2p connectivity.
+In case the sync port (3015 by default) cannot be used as external forwarding port the `sync` > `external_port` configuration parameter can be used to change it. This is the port that the node will advertise to the network to be reached over Internet.
+
+Example scheme: `node (sync > port) <-> router (sync > external_port) <-> Internet`
+
+#### Port Check
+
+In order to validate your setup and configuration correctness one can run the external node port check (assuming the default port 3015):
```

Comment:
I just tried this, and it is very important to mention:
"After you have started the epoch application, you can verify the validity of your setup and configuration correctness by, for example, running the external node port check (assuming the default port 3015".

I tried before starting the application and clearly the port was still closed.